chain/src/auxdata.rs

```diff
@@ -0,0 +1,77 @@
+/// TX auxiliary data
+use super::*;
```

Comment:
This is the old `chain/src/metadata.rs` but with the metadata stuff removed and moved to `core/src/metadata.rs`